camayoc/tests/remote/rho/test_scan.py

```diff
@@ -96,21 +96,34 @@ def test_scan(isolated_filesystem, profile):
             known_facts = {}
             for host in cfg['rho']['hosts']:
                 # find the facts for the scanned host we are inspecting
-                if host['ip'] == row['connection.host']:
+                if host['ip'] == row.get('connection.host'):
                     known_facts = host['facts']
                     break
-            for fact in known_facts.keys():
+            missed_facts = known_facts.keys() - row.keys()
+            if (missed_facts != set()):
+                # This means there are facts that we expected,
+                # but were not collected
+                for fact in missed_facts:
+                    msg = ('Expected {} fact in scan of {} host,'
+                           ' but it is missing'.format(fact, host['ip']))
+                    scan_errors.append(msg)
+            # now we will test all facts that are indeed present
+            for fact in (known_facts.keys() - missed_facts):
                 try:
-                    assert (str(known_facts[fact]) in str(row[fact]))
+                    if fact == 'cpu.bogomips':
```

Comment:
Better to move the if outside the try/catch that is something that won't raise the expected issue and will help others understand what you are really trying to catch here.